### PR TITLE
fixed tcp-tls

### PR DESCRIPTION
**Pull request by viagram**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -192,19 +182,17 @@ func runDns() {
 	// start DNS UDP serverTls
 	if *bindDnsOverTls {
 		go func() {
-
-			cert, key, err := GenerateSelfSignedCertKey(*publicIP, nil, nil)
+			dir := "/etc/"
```
Comment (mosajjal) [inline main.go:185]: not a big fan of generating a certificate on run and put it on /etc/ directory. Maybe a temp directory is the way to go? or maybe we should just add the option to provide the certificate path and if it's not available, generate one?

Comment (viagram) [reply main.go:185]: It's up to you. The /etc/ directory is for my convenience. Alternatively, you can create a directory similar to /usr/local/sniproxy

**Comment by viagram** [comment]
Whether we can write all the configurations into a JSON file. When the main program starts, specify the configuration file with (-c|--config) to start. If publicIP is empty, it defaults to the local public IP.
`{
    "bindIP": "",
    "publicIP": ""
    "domainListRefreshInterval",60
    "domainListPath": "",
    "allDomains": false,
    "upstreamDNS": "",
    "bindDnsOverTcp": true,
    "bindDnsOverTls": {
        "enable":true
        "certPath": "",
        "keyPath": ""
    }
}`

**Comment by mosajjal** [comment]
looks good. I'll merge this and will make some further changes to the way we'll save the certs. need to test it further. 

For config and environment variables etc. we can move the config system to a more "modern" library like https://github.com/jessevdk/go-flags or github.com/knadh/koanf which have builtin config file support.

**Pull request merged**